### PR TITLE
init cli: add option for installing package

### PR DESCRIPTION
**Pull request by gtpan77**
Closes: #3582

Added options for installing package either using npm or yarn by providing `--pm` or `--manager` option

### CODE REVIEW

File: src/commands/InitCommand.ts [hunk 1]
```diff
@@ -81,11 +93,22 @@ export class InitCommand implements yargs.CommandModule {
     // Protected Static Methods
     // -------------------------------------------------------------------------
 
+    protected static executeCommand(command: string) {
+        return new Promise<string>((ok, fail) => {
+            exec(command, (error: any, stdout: any, stderr: any) => {
+                if (stdout) return ok(stdout);
+                if (stderr) return ok(stderr);
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:100]: Wouldn't `stderr` then a failure exit code cause this to not reject?  Because it would resolve first and then never reject.

File: src/commands/InitCommand.ts [hunk 2]
```diff
@@ -202,7 +225,7 @@ export class InitCommand implements yargs.CommandModule {
                 sourceMap: true
             }
         }
-        , undefined, 3);
+            , undefined, 3);
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:228]: Can you undo this change?

File: src/commands/InitCommand.ts [hunk 3]
```diff
@@ -221,13 +244,13 @@ temp/`;
...
-        return `import {Entity, ${ database === "mongodb" ? "ObjectIdColumn, ObjectID" : "PrimaryGeneratedColumn" }, Column} from "typeorm";
+        return `import {Entity, ${database === "mongodb" ? "ObjectIdColumn, ObjectID" : "PrimaryGeneratedColumn"}, Column} from "typeorm";
 
 @Entity()
 export class User {
 
-    ${ database === "mongodb" ? "@ObjectIdColumn()" : "@PrimaryGeneratedColumn()" }
-    id: ${ database === "mongodb" ? "ObjectID" : "number" };
+    ${database === "mongodb" ? "@ObjectIdColumn()" : "@PrimaryGeneratedColumn()"}
+    id: ${database === "mongodb" ? "ObjectID" : "number"};
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:253]: Can you undo thee changes?

File: src/commands/InitCommand.ts [hunk 4]
```diff
@@ -33,6 +34,10 @@ export class InitCommand implements yargs.CommandModule {
             })
             .option("docker", {
                 describe: "Set to true if docker-compose must be generated as well. False by default."
+            })
+            .option("pm", {
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:38]: Can we instead just make a flag for `yarn` and install with `npm` by default?  What does webpack do?

File: src/commands/InitCommand.ts [hunk 5]
```diff
@@ -33,6 +34,10 @@ export class InitCommand implements yargs.CommandModule {
...
+                alias: "manager",
+                describe: "Install packages, expected values are npm or yarn."
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:40]: If these are the valid values, should we set `choices` to these two?

File: src/commands/InitCommand.ts [hunk 6]
```diff
@@ -70,6 +76,12 @@ export class InitCommand implements yargs.CommandModule {
                 console.log(chalk.green(`Project created inside current directory.`));
             }
 
+            if (args.pm && installNpm) {
+                InitCommand.executeCommand("npm install");
+            } else {
+                InitCommand.executeCommand("yarn install");
```
Comment (imnotjames) [inline src/commands/InitCommand.ts:82]: Shouldn't you `await` this?  Or we don't wait for them to complete ever?

**Review by imnotjames — CHANGES_REQUESTED** [review]

File: src/commands/InitCommand.ts (hunk 2 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:228]: Sure, but it is not done by me. VScode auto formatter make this change on saving the file. This change will not affect the code.

File: src/commands/InitCommand.ts (hunk 3 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:253]: same here, these changes are only formatting. It will not affect the functioning of the code.

File: src/commands/InitCommand.ts (hunk 4 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:38]: I thought of making npm default option if flag is not provided. We can make one flag for yarn installation only. Need your confirmation here.

File: src/commands/InitCommand.ts (hunk 6 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:82]: my bad. `await` should be there.

File: src/commands/InitCommand.ts (hunk 1 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:100]: Hmm.. I took this implementation from `VersionCommand.ts`. Should I call fail for `stderr`?

File: src/commands/InitCommand.ts (hunk 5 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:41]: done

File: src/commands/InitCommand.ts (hunk 4 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:38]: @imnotjames, need your final confirmation here.

**Comment by pleerock** [comment]
I'm sorry for so late review, just found a time to review it.
Are you still interested in this change? Looks like imnotjames don't have a time currently to answer your questions. Let me review it.

File: src/commands/InitCommand.ts (hunk 4 above)
Comment (pleerock) [reply src/commands/InitCommand.ts:38]: default should be npm, if users use `init` command without `pm` option specified - it should work and default should be npm.

File: src/commands/InitCommand.ts (hunk 1 above)
Comment (pleerock) [reply src/commands/InitCommand.ts:101]: Yes, I think `fail` should be called here. Also, probably `VersionCommand` is incorrect.

**Comment by gtpan77** [comment]
Yes, I'm interested in these changes.

File: src/commands/InitCommand.ts (hunk 1 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts]: done

File: src/commands/InitCommand.ts (hunk 4 above)
Comment (gtpan77) [reply src/commands/InitCommand.ts:38]: done

**Comment by AlexMesser** [comment]
thank you for contribution!
